owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/MainFileListFragment.kt

```diff
@@ -798,10 +798,15 @@ class MainFileListFragment : Fragment(),
         collectLatestLifecycleFlow(mainFileListViewModel.fileListUiState) { fileListUiState ->
             if (fileListUiState !is MainFileListViewModel.FileListUiState.Success) return@collectLatestLifecycleFlow
 
+            saveScrollPosition()
+
             fileListAdapter.updateFileList(
                 filesToAdd = fileListUiState.folderContent,
                 fileListOption = fileListUiState.fileListOption,
             )
+
+            restoreScrollPosition()
+
```

Comment:
Hmmm I don't know if I like this to be here... there are 2 main points that make me doubt here:
1. The observer where this is located. This is going to be executed everytime `fileListUiState` is updated, and that is a `combine` with 5 different variables: the current folder displayed, the file list option, the search filter, the sort type and order and the current space. Everytime one of these is updated, the scroll position is going to be saved (and the previous one restored), so it looks a bit inefficient (and a potential source of glitches)
2. Seems we're saving just 1 scroll position, but what if we go deep into a several-level folder hierarchy? Would we be able to restore every scroll position of each level?